python/src/dynamixel_sdk/protocol1_packet_handler.py

```diff
@@ -131,7 +139,7 @@ def rxPacket(self, port, rxpacket=[]):
         wait_length     = 6 # minimum length (HEADER0 HEADER1 ID LENGTH ERROR CHKSUM)
 
         while True:
-            rxpacket += port.readPort(wait_length - rx_length)
```

Comment:
subtle bug. if += actually means x = x + y, this is a reassignment to parameter label. This means that you loose the relationship with the outer element passed as parameter

Reply:
@roncapat 

well, rxpacket=[] is let opened to make the parameter format same as what cpp has. So the parameter in the code actually is shown like it uses the passed parameter, but in the other functions which uses rxPacket(), it doesn't get any parameter for rxpacket.

or do you still think that it doesn't need to get the rxpacket at first?

Reply:
I mean, in C++ rxpacket is meant as a return parameter, so something where you put the results. But in this python code, if you do another assignment to the local rxpacket label, you would overwrite the reference it has to the passed argument. So when you return from the function, the list you passed would not have the results you expected to have there.

Reply:
By extending it, instead, I add elements to the same list, I do not overwrite it with a new list. That's, in theory, the difference. It should work.

Reply:
@roncapat 

oops... that window over there doesn't show the next line, so I couldn't read your idea before I found out from file changes. Understood.

Reply:
@roncapat 

looks like append should be replaced with extend... it bugs

Reply:
You're right